package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -818,23 +842,51 @@ export const MessageInputProvider = <
     }
   };
 
+  const regExcondition = /File (extension \.\w{2,4}|type \S+) is not supported/;
+
+  const setFileUploadState =
```

Comment:
I think this function uses currrying to set the upload state of the file. I also don't understand this function well enough cause I didn't write this it and I fear rewriting it might lead me into another rabbit hole. Maybe we can rewrite this is a different PR

Reply:
Thanks @santhoshvai, after going through this function again and understanding it better I have renamed it to fit what it does. The currying concept I agree is confusing but I suggest we improve it in a different PR, perhaps we open a technical depth issue for this?

Reply:
In my view, the currying concept is only confusing to the reader if the naming of the variable is unclear, not necessarily confusing.